modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/ValidationHelper.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.headless.builder.internal.helper;
+
+import com.liferay.headless.builder.application.APIApplication;
+import com.liferay.object.exception.ObjectEntryValuesException;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.rest.dto.v1_0.ListEntry;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.io.Serializable;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alberto Javier Moreno Lage
+ */
+@Component(service = ValidationHelper.class)
+public class ValidationHelper {
+
+	public boolean isValidObjectEntry(
+			long objectEntryId, String externalReferenceCode)
+		throws Exception {
+
+		if (objectEntryId == 0) {
+			return false;
+		}
+
+		ObjectEntry objectEntry = _objectEntryLocalService.fetchObjectEntry(
+			objectEntryId);
+
+		if (objectEntry == null) {
+			return false;
+		}
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectEntry.getObjectDefinitionId());
+
+		if (!Objects.equals(
+				objectDefinition.getExternalReferenceCode(),
+				externalReferenceCode)) {
+
+			return false;
+		}
+
+		return true;
+	}
+
+	public void validateAPIEndpointRelationship(
```

Comment:
Or if this `ValidationHelper` is related to an specific entity in API Builder which I can see it is `APIEndpointValidatorHelper`. What do you think @4lejandrito?

Reply:
I don't know about this. I get the point that abstraction could bring great benefits in some points of the code, but I think there's a limit. The point is that this validation helper is created within the API Builder module so it makes sense that it's purpose is for validating different API Builder entities.

Also I think this class isn't related to an specific entity since the method `isValidObjectEntry` is used by API Endpoints, API Schema and API Properties (also by Sorts and Filters but within the method found inside the class).

Still, I'll wait to see what @4lejandrito thinks about this.

Reply:
Ideally it should be just `validateRelationshiop` and we should pass the name of the object definitions as parameters. But that is what caused brian to reject this in the first place... so I think it is good enough for now.